Proxy settings fixes

galeksandrp opened the pull request:
- Fixes `Proxy from Elementum settings` setting

Log reported setting applied without really doing anything.

- Adds `Proxy from Elementum (resolve hosts by proxy)` setting option

Option enables default behaviour of `Elementum` in `Burst`. Option is set as new default,
existing users should not be affected by change.

- Adds choice of additional proxy protocols in settings

`SOCKS4a` (`SOCKS4` latest version with hostname resolve by proxy)
`SOCKS5h` (`SOCKS5` latest version with hostname resolve by proxy). Protocol is set as new
default to match default setting in `Elementum`, existing users should not be affected by
change.

- Adds help label for `use_elementum_proxy` setting

elgatito commented:
@galeksandrp Was that testes with Kodi 17/18?

galeksandrp commented:
Version | Operating System | Result
--- | --- | ---
`Kodi 17.6 Git:20171114-a9a7a20` | `Windows 10` `2004` | ALL OK <details>✅Proxying `Burst` via `Elementum`<br>✅Using proxy settings from `Elementum`<br>✅Proxy settings from `Elementum` and hosts resolve by proxy<br>✅Using `SOCKS5h` proxy in `Burst`<br>✅Using `SOCKS4a` proxy in `Burst`<br>✅Using `SOCKS5` proxy in `Burst`<br>✅Using `SOCKS4` proxy in `Burst`</details>
`Kodi 18.9 (18.9.0) Git:20201023-0655c2c718` | `Windows 10` `2004` | ALL OK <details>✅Proxying `Burst` via `Elementum`<br>✅Using proxy settings from `Elementum`<br>✅Proxy settings from `Elementum` and hosts resolve by proxy<br>✅Using `SOCKS5h` proxy in `Burst`<br>✅Using `SOCKS4a` proxy in `Burst`<br>✅Using `SOCKS5` proxy in `Burst`<br>✅Using `SOCKS4` proxy in `Burst`</details>
`19.1 (19.1.0) Git: 20210508-85e05228b4` | `Windows 10` `2004` | ALL OK <details>✅Proxying `Burst` via `Elementum`<br>✅Using proxy settings from `Elementum`<br>✅Proxy settings from `Elementum` and hosts resolve by proxy<br>✅Using `SOCKS5h` proxy in `Burst`<br>✅Using `SOCKS4a` proxy in `Burst`<br>✅Using `SOCKS5` proxy in `Burst`<br>✅Using `SOCKS4` proxy in `Burst`</details>

antonsoroko commented on resources/language/messages.pot:
```diff
@@ -244,7 +244,7 @@ msgid "Use proxy settings"
 
 msgctxt "#32107"
-msgid "Proxy from Elementum settings"
+msgid "From Elementum settings"
```
@galeksandrp 
since you changed template - you also need to change all translated files, otherwise the will lose old translation after running synchronization scripts in `scripts` dir.

`for file in resources/language/*/*.po; do sed -i 's/msgid "Proxy from Elementum settings"/msgid "From Elementum settings"/' "$file"; done`
will do the trick.

galeksandrp replied:
Done, thanks!

elgatito commented:
@galeksandrp please, run 
```
./scripts/xgettext.sh
./scripts/xgettext_merge.sh
```

Is this finished? Working as expected?

elgatito commented:
@galeksandrp ping

galeksandrp commented:
<details>
<summary>./scripts/xgettext.sh</summary>

```
[galek@HP2 script.elementum.burst]$ ./scripts/xgettext.sh 
Checking resources/language/Croatian/strings.po                        [  OK  ]
Checking resources/language/English/strings.po                         [  OK  ]
Checking resources/language/French/strings.po                          [  OK  ]
Checking resources/language/German/strings.po                          [  OK  ]
Checking resources/language/Hebrew/strings.po                          [  OK  ]
Checking resources/language/Hungarian/strings.po                       [  OK  ]
Checking resources/language/Italian/strings.po                         [  OK  ]
Checking resources/language/Lithuania/strings.po                       [  OK  ]
Checking resources/language/Portuguese (Brazil)/strings.po             [  OK  ]
Checking resources/language/Portuguese/strings.po                      [  OK  ]
Checking resources/language/Russian/strings.po                         [  OK  ]
Checking resources/language/Spanish (Argentina)/strings.po             [  OK  ]
Checking resources/language/Spanish (Mexico)/strings.po                [  OK  ]
Checking resources/language/Spanish/strings.po                         [  OK  ]
Checking resources/language/Swedish/strings.po                         [  OK  ]
Checking resources/language/Ukrainian/strings.po                       [  OK  ]
[galek@HP2 script.elementum.burst]$ echo $?
0
[galek@HP2 script.elementum.burst]$ git status
On branch socks-hostname-resolve
Your branch is up to date with 'origin/socks-hostname-resolve'.

nothing to commit, working tree clean
```
</details>

<details>
<summary>./scripts/xgettext_merge.sh</summary>

```
[galek@HP2 script.elementum.burst]$ ./scripts/xgettext_merge.sh
Merging resources/language/Croatian/strings.po                          
[  OK  ]

Merging resources/language/English/strings.po                           
[  OK  ]

Merging resources/language/French/strings.po                            
[  OK  ]

Merging resources/language/German/strings.po                            
[  OK  ]

Merging resources/language/Hebrew/strings.po                            
[  OK  ]

Merging resources/language/Hungarian/strings.po                         
[  OK  ]

Merging resources/language/Italian/strings.po                           
[  OK  ]

Merging resources/language/Lithuania/strings.po                         
[  OK  ]

Merging resources/language/Portuguese (Brazil)/strings.po               
[  OK  ]

Merging resources/language/Portuguese/strings.po                        
[  OK  ]

Merging resources/language/Russian/strings.po                           
[  OK  ]

Merging resources/language/Spanish (Argentina)/strings.po               
[  OK  ]

Merging resources/language/Spanish (Mexico)/strings.po                  
[  OK  ]

Merging resources/language/Spanish/strings.po                           
[  OK  ]

Merging resources/language/Swedish/strings.po                           
[  OK  ]

Merging resources/language/Ukrainian/strings.po                         
[  OK  ]

[galek@HP2 script.elementum.burst]$ echo $?
0
[galek@HP2 script.elementum.burst]$ git status
On branch socks-hostname-resolve
Your branch is up to date with 'origin/socks-hostname-resolve'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   resources/language/Croatian/strings.po
        modified:   resources/language/French/strings.po
        modified:   resources/language/German/strings.po
        modified:   resources/language/Hebrew/strings.po
        modified:   resources/language/Hungarian/strings.po
        modified:   resources/language/Italian/strings.po
        modified:   resources/language/Lithuania/strings.po
        modified:   resources/language/Portuguese (Brazil)/strings.po
        modified:   resources/language/Portuguese/strings.po
        modified:   resources/language/Russian/strings.po
        modified:   resources/language/Spanish (Argentina)/strings.po
        modified:   resources/language/Spanish (Mexico)/strings.po
        modified:   resources/language/Spanish/strings.po
        modified:   resources/language/Swedish/strings.po
        modified:   resources/language/Ukrainian/strings.po

no changes added to commit (use "git add" and/or "git commit -a")
```
</details>

I am committed files changed by `./scripts/xgettext_merge.sh` into separate commit, you can merge PR` with `squash`.
